### DISCUSSION TITLE
What is y'all's approach to fullscreen games that utilize a lot of the keyboard?

### DISCUSSION THREAD

**Discussion by ELLIOTTCABLE** [nikitabobko/AeroSpace#679] [questions-and-answers] [answered]
For example, at the moment, I'm playing Factorio. There's a lot of complex keybinds, especially (and unusually amongst games, in my experience) a lot that use <kbd>Alt</kbd>, that overlap with my Aerospace config substantially.

Various fixes for this include:

1. (terrible) Remap _all_ of of a game's offending keys (quite arduous, and even potentially nearly impossible if there's enough operations that need to stay close to the homerow while playing …);

2. (terrible) quit Aerospace when you're going to play;

3. (bad) manually switch to a mode/profile that disables most operations while playing, then switch back to `main` when you want your keybinds restored;

4. (less bad) switch to a mode with less functionality automatically when one of <some set of games> is launched;

5. (maybe best?) configure some games to launch into native macOS fullscreen, and (I don't even know if Aerospace can do this?) disable Aerospace's keybinds when the native-fullscreen game is focused?

None of these seem easy; what have y'all been doing?

**Comment by alextremblay** [comment]
seems to me like the best option would be to disable aerospace while factorio is focused.

maybe you could implement that using an `on-focus-changed` callback?

**Comment by gldtn** [comment, marked as answer]
> 3\. (bad) manually switch to a mode/profile that disables most operations while playing, then switch back to `main` when you want your keybinds restored;

I suggest to use a slightly different approach. Use a vim-like config: https://github.com/nikitabobko/AeroSpace/discussions/620

```toml
[mode.main.binding]
ctrl-space = ['mode alt']

# 'alt' binding mode declaration
[mode.alt.binding]
esc = ['mode main']
```

Then set all your aerospace binding in the `alt binding` mode

This way should give you more flexibility and less chance of key binding collision

**Reply by ELLIOTTCABLE** [reply]
That's a really good idea, actually. A bit like Tridactyl's [`:mode ignore`](https://tridactyl.xyz/build/static/docs/modules/_src_excmds_.html#mode). I think I'll steal that, thank you!
